game/graphics/opengl_renderer/DirectRenderer.h

```diff
@@ -237,13 +244,42 @@ class DirectRenderer : public BucketRenderer {
     void push(const math::Vector<u8, 4>& rgba,
               const math::Vector<u32, 4>& vert,
               const math::Vector<float, 3>& stq,
+              const math::Vector<float, 4>& scissor,
               int unit,
               bool tcc,
               bool decal,
               bool fog_enable,
               bool use_uv);
   } m_prim_buffer;
 
+  // the scissor state tends to be shared across buckets, so it is static here
```

Comment:
this seems kinda suspicious, but if it becomes a problem we can just flush everything on scissor state changes.